{
  final String nil=null;
  try {
    nil.toString();
    Assert.fail();
  }
 catch (  final NullPointerException e) {
    final BillingExceptionJson exceptionJson=new BillingExceptionJson(e,true);
    Assert.assertEquals(exceptionJson.getClassName(),e.getClass().getName());
    Assert.assertNull(exceptionJson.getCode());
    Assert.assertNull(exceptionJson.getMessage());
    Assert.assertNull(exceptionJson.getCauseClassName());
    Assert.assertNull(exceptionJson.getCauseMessage());
    Assert.assertFalse(exceptionJson.getStackTrace().isEmpty());
    Assert.assertEquals(exceptionJson.getStackTrace().get(0).getClassName(),TestBillingExceptionJson.class.getName());
    Assert.assertEquals(exceptionJson.getStackTrace().get(0).getMethodName(),"testFromException");
    Assert.assertFalse(exceptionJson.getStackTrace().get(0).getNativeMethod());
    final BillingExceptionJson exceptionJsonNoStackTrace=new BillingExceptionJson(e,false);
    Assert.assertEquals(exceptionJsonNoStackTrace.getClassName(),e.getClass().getName());
    Assert.assertNull(exceptionJsonNoStackTrace.getCode());
    Assert.assertNull(exceptionJsonNoStackTrace.getMessage());
    Assert.assertNull(exceptionJsonNoStackTrace.getCauseClassName());
    Assert.assertNull(exceptionJsonNoStackTrace.getCauseMessage());
    Assert.assertTrue(exceptionJsonNoStackTrace.getStackTrace().isEmpty());
  }
}
